frankenstein/chunk_store.go

```diff
@@ -367,34 +390,43 @@ func (c *AWSChunkStore) lookupChunks(userID string, from, through model.Time, ma
 		go c.lookupChunksFor(userID, hour, metricName, matchers, incomingChunkSets, incomingErrors)
 	}
 
-	chunkSets := []wire.Chunk{}
+	chunks := []wire.Chunk{}
 	errors := []error{}
 	for i := 0; i < len(buckets); i++ {
 		select {
-		case chunkSet := <-incomingChunkSets:
-			chunkSets = append(chunkSets, chunkSet...)
+		case incoming := <-incomingChunkSets:
+			chunks = merge(chunks, incoming)
 		case err := <-incomingErrors:
 			errors = append(errors, err)
 		}
 	}
 	if len(errors) > 0 {
 		return nil, errors[0]
 	}
-	return chunkSets, nil
+	return chunks, nil
+}
+
+func next(s string) string {
+	// TODO deal with overflows
+	l := len(s)
+	return s[:l-2] + string(s[l-1]+1)
 }
 
 func (c *AWSChunkStore) lookupChunksFor(userID string, hour int64, metricName model.LabelValue, matchers []*metric.LabelMatcher, incomingChunkSets chan []wire.Chunk, incomingErrors chan error) {
 	var chunkSets [][]wire.Chunk
 	for _, matcher := range matchers {
-		// TODO build support for other matchers
-		if matcher.Type != metric.Equal {
-			incomingErrors <- fmt.Errorf("%s matcher not supported yet", matcher.Type)
-			return
-		}
 
 		hashValue := hashValue(userID, hour, metricName)
-		rangeMinValue := rangeValue(matcher.Name, matcher.Value, minChunkID)
-		rangeMaxValue := rangeValue(matcher.Name, matcher.Value, maxChunkID)
+		var rangeMinValue, rangeMaxValue []byte
+		if matcher.Type == metric.Equal {
+			nextValue := model.LabelValue(next(string(matcher.Value)))
+			rangeMinValue = rangeValue(matcher.Name, matcher.Value, "")
+			rangeMaxValue = rangeValue(matcher.Name, nextValue, "")
```

Comment:
Yes I hope so; just need to check if the ranges are inclusive or not.  Its currently not quite working as expected.